### PR TITLE
Adjust conversion from nuint to float/double

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/60714
Relevant spec: https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/native-integers.md#conversions

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs [hunk 1]
```diff
@@ -14687,6 +14738,92 @@ enum E { }
             }
         }
 
+        [Fact, WorkItem(60714, "https://github.com/dotnet/roslyn/issues/60714")]
+        public void ConversionFromNuintToDouble()
+        {
+            var source = """
+if (!System.Environment.Is64BitProcess)
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs:14745]: Is there an equivalent test that fails on 32-bit platforms? 

For instance, consider changing the code below to:
```csharp
nuint x = System.IntPtr.Size == 4 ?
    uint.MaxValue :
    unchecked((nuint)ulong.MaxValue);
ulong y = System.IntPtr.Size == 4 ?
    uint.MaxValue :
    ulong.MaxValue;
```

File: src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs [hunk 2]
```diff
@@ -14687,6 +14738,92 @@ enum E { }
...
+  IL_0002:  conv.r8
+  IL_0003:  ret
+}");
+        }
+
+        [Fact, WorkItem(60714, "https://github.com/dotnet/roslyn/issues/60714")]
+        public void ConversionFromNuintToFloat()
+        {
+            var source = """
+if (!System.Environment.Is64BitProcess)
```
Comment (cston) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs:14788]: We should test `uint.MaxValue` on 32-bit platforms.

**Review by cston — APPROVED** [review]

File: src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs]: Updated and verified those two tests fails both on 32-bit and 64-bit platforms prior to fix.

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for a second review. Thanks

**Comment by jcouv** [comment]
@333fred @dotnet/roslyn-compiler for a second review. Thanks

**Comment by jcouv** [comment]
@333fred @dotnet/roslyn-compiler for a second review. Thanks

**Review by 333fred — APPROVED** [review]

**Pull request merged**